Document the `pubspec_overrides.yaml` file (#6462)
dart-lang/site-www: src/content/tools/pub/dependencies.md

keep in mind that your package's dependency overrides
are ignored by all users of your package.

> If you are using a [pub workspace][workspaces], you can place `dependency_overrides` in all
> workspace packages, but a single package can only be overridden once in the
> workspace.

Reviewer: ```suggestion
If you are using a [pub workspace][workspaces], you can
have `dependency_overrides` in each workspace package, but
a single package can only be overridden once in the workspace.
```

## `pubspec_overrides.yaml`

If you want to change certain aspects of your `pubspec.yaml` but do not

> want to have active changes to the actual file, you can place a file called

Reviewer: Could you expand on what you meant by active in "active changes to the actual file"?

Do you mean changes that would affect consumers of the package?

Author: I mean something like "active version control changes" but I tried to simplify, just focus on avoiding changing the main file.

`pubspec_overrides.yaml` placed next to `pubspec.yaml`.

Attributes from that file will override those from `pubspec.yaml`.

The properties that can be overridden are:

* `dependency_overrides`
* `workspace`
* `resolution`

This can be useful to avoid accidentally checking temporary overrides in to 

> version control. Also it can make it easier to generate overrides from a script. 

Reviewer: ```suggestion
version control. It can also make it easier to generate overrides from a script. 
```
